templates/common/_base/files/wait-for-primary-ip.yaml

```diff
@@ -0,0 +1,31 @@
+mode: 0755
+path: "/usr/local/bin/wait-for-primary-ip.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -eux
+
+    if [ ! -e /etc/nmstate/openshift/applied ]; then
+      # No need to do this if no NMState configuration was applied
+      exit 0
+    fi
+
+    # This logic is borrowed from configure-ovs.sh
```

Comment:
I guess this makes the check in configure-ovs dead code?

Reply:
Not in this case. We only run this if the new mechanism is used, so the configure-ovs version will still apply when we don't.

Reply:
Ah ok. Why do you need to use `/run/nodeip-configuration/primary-ip` specifically? If configure-ovs is doing it and it is platform agnostic, then we could basically use whatever it is using. I guess not problem if we limit this to baremetal though.

Reply:
I think this couldn't work exactly the same way configure-ovs does because it's not all one service. In this case, we have nmstate-configuration.service providing configuration files for nmstate.service, then run wait-for-primary-ip.service to ensure we have a usable address.

The platform-agnostic solution is probably to split this into two services, one that runs before nmstate.service and stores off the primary IP, then this one that runs after to wait for it. I'll see how hard that would be to do now that we have a little more time to get this done.